test/unit/plugins/rest_spec.js

```diff
@@ -646,4 +647,97 @@ describe("The Rest plugin", function () {
 			});
 		});
 	});
+
+	describe("updating a created instance", function () {
+
+		var server    = new Hapi.Server();
+		var instances = new InstanceAdapter();
+		var result;
+		var instanceID;
+		var updatedInstance;
+
+		before(function () {
+
+			server.connection();
+			return server.registerAsync({
+				register : Rest,
+				options  : {
+					instances : instances
+				}
+			});
+		});
+
+		after(function () {
+			return server.stopAsync();
+		});
+
+		describe("setting the status of a created instance to failed", function () {
+
+			before(function () {
+				return instances.createInstance()
+				.then(function (instance) {
+					instanceID = instance.id;
+					updatedInstance = new Instance({
+						id    : instance.id,
+						ami   : instance.ami,
+						type  : instance.type,
+						state : "failed",
+						uri   : instance.uri
+					});
+
+					var request = new Request("PUT", "/v1/instances/" + instance.id).mime("application/json")
+					.payload({
+						ami   : instance.ami,
+						type  : instance.type,
+						uri   : instance.uri,
+						state : "failed"
+					});
+					return request.inject(server);
+				})
+				.then(function (response) {
+					console.log(response);
+					result = response.result;
+				});
+			});
+
+			it("the status is set to failed", function () {
+				return instances.getInstance({ id : instanceID })
+				.then(function () {
+					expect(result.id).to.equal(updatedInstance.id);
+					expect(result.ami).to.equal(updatedInstance.ami);
+					expect(result.type).to.equal(updatedInstance.type);
+					expect(result.state).to.equal(updatedInstance.state);
+					expect(result.uri).to.equal(updatedInstance.uri);
+				});
+			});
+		});
+
+		describe("when the instance ID doesn't exist", function () {
+
+			before(function () {
+				return instances.createInstance()
+				.then(function (instance) {
+					var request = new Request("PUT", "/v1/instances/" + INVALID_QUERY).mime("application/json")
+					.payload({
+						ami   : instance.ami,
+						type  : instance.type,
+						uri   : instance.uri,
+						state : "failed"
+					});
+					return request.inject(server);
+				})
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("the status is set to failed", function () {
+				return instances.getInstance({ id : INVALID_QUERY })
+				.then(function () {
+					var error = JSON.parse(result.payload);
```

Comment:
If you save the entire `response` object above, then you shouldn't need to parse the payload here. Instead, you can inspect `response.statusCode`. In fact, this is what we actually want to check since the HTTP status is not necessarily related to the payload.